pkg/cache/cache.go

```diff
@@ -657,6 +661,41 @@ func filterLocalQueueUsage(orig FlavorResourceQuantities, resourceGroups []Resou
 	return qFlvUsages
 }
 
+func workloadOrderingFunc(a, b *workload.Info) bool {
+	p1 := utilpriority.Priority(a.Obj)
+	p2 := utilpriority.Priority(b.Obj)
+
+	if p1 != p2 {
+		return p1 > p2
+	}
+
+	return a.Obj.CreationTimestamp.Before(&b.Obj.CreationTimestamp)
+}
+
+func (c *Cache) RunningWorkload(name string) ([]*workload.Info, error) {
+	c.RLock()
+	defer c.RUnlock()
+	cq, ok := c.clusterQueues[name]
+	if !ok {
+		return []*workload.Info{}, apierr.NewNotFound(v1alpha1.Resource("clusterqueue"), name)
+	}
+
+	count := 0
+	wkls := make([]*workload.Info, 0, len(cq.Workloads))
+	for k, wkl := range cq.Workloads {
+		if _, ok := c.assumedWorkloads[k]; ok {
```

Comment:
I am not sure about this condition as well, could you clarify the intentions please? Another way of checking if a workload is running would be checking its conditions - it's accessible in `wkl` object here